csrc/ops/arith.cpp

```diff
@@ -1478,11 +1479,10 @@ TensorView* expand(TensorView* inp, const std::vector<Val*>& expanded_sizes) {
       // already done when constructing out_id_builder.
       out_id_builder.extent(inp_id->extent());
     } else if (
-        (inp_id->isBroadcast() ||
-         // special patch for Symbolic IterDomain with a static size-1 extent
-         // See Issue: https://github.com/NVIDIA/Fuser/pull/1393
-         (inp_id->isSymbolic() && inp_id->extent()->isConstInt() &&
-          inp_id->extent()->evaluate() == 1)) &&
+        // special patch for Symbolic IterDomain with a static size-1 extent
+        // since we know it will become broadcast at concretization
+        // See Issue: https://github.com/NVIDIA/Fuser/pull/1393
+        (inp_id->extent()->isConstInt() && inp_id->extent()->evaluate() == 1) &&
```

Comment:
Does this mean we need to update our `isBroadcast()` implementation to use this notion `(inp_id->extent()->isConstInt() && inp_id->extent()->evaluate() == 1)` instead?

With symbolic_iterDomain, clearly the `isBraodcast()` is no longer sufficient any more.

Reply:
Until #1393 I thought we would not have `Symbolic` ID with size-1 extent, but then we did. It seems like expanding a symbolic domain to constant size 1 is the only way for that to happen, so we should probably more carefully check that particular case here and ensure the output will be set as `Broadcast` (I don't think it will be in the current version here).

I don't think we should modify `isBroadcast()`. `Symbolic` can still be either `Iteration` or `Broadcast` and `isBroadcast` only means it's guaranteed to be broadcast.

Reply:
Yeah I agree with you on keeping `isBroadcast` as-is (a guaranteed broadcast).

Does it make sense to add a `maybeBroadcast` as a simpler API to address the ambiguity introduced by symbolic iterDomain?

Reply:
Something like the following?
```c++
namespace ir_utils {
bool maybeBroadcast(IterDomain* id) {
  return id->isBroadcast() || id->isSymbolic();
}
} // namespace ir_utils
```
Sounds good but the name in this case is not quite as precise as the explicit condition, so it might be a little confusing :man_shrugging: ?

Reply:
I'm bad with names. if you think it's a confusing name, I'll trust your judgement on that one :laughing: 

please just ignore the previous request. sorry about the noise.